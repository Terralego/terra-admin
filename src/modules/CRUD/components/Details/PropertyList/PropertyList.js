import React from 'react';
import PropTypes from 'prop-types';
import PropertyItem from '../PropertyItem';

const PropertyList = ({ properties }) => (
  <ul className="details__list">
    {Object.entries(properties).map(([key, value]) => (
      <PropertyItem
        key={key}
        name={key}
        value={value}
      />
    ))}
  </ul>
);

PropertyList.propTypes = {
  properties: PropTypes.shape({}),
};

PropertyList.defaultProps = {
  properties: {},
};

export default PropertyList;
