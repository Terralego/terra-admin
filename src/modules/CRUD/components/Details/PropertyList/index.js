import PropertyList from './PropertyList';

export default PropertyList;
