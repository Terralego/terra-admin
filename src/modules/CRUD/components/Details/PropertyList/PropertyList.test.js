import React from 'react';
import renderer from 'react-test-renderer';

import PropertyList from './PropertyList';

jest.mock('./PropertyList', () => props => <div {...props} />);

const props = {
  properties: {
    title: 'Title of the feature',
    date: '2020/02/21',
  },
};

it('should render correctly', () => {
  const tree = renderer.create((
    <PropertyList {...props} />
  )).toJSON();
  expect(tree).toMatchSnapshot();
});
