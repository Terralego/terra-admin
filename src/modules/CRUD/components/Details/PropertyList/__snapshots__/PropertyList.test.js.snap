// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  properties={
    Object {
      "date": "2020/02/21",
      "title": "Title of the feature",
    }
  }
/>
`;
